app/controllers/prison/print_visits_controller.rb

```diff
@@ -27,6 +27,7 @@ def create_visit_date
     @submitted_date = AccessibleDate.new(
       date_to_accessible_date(@print_visits.visit_date)
       ).to_date
+    @print_visits.check_date(@submitted_date.strftime('%F')) unless @submitted_date.nil?
```

Comment:
This method is doing more than just creating a date ... should it's name be updated accordingly (or perhaps move the changes to its own method ...)

Reply:
Rather than calling 'check_date?' shouldn't the validation be invoked via 'valid?'

Reply:
here we should just call `@print_visits.valid?` that @check_date can be called internally by the model since it has access to the date. So no need to pass stuff around